### PR TITLE
Renamed all exercise example files, added support for concept exercise tests

### PR DESCRIPTION
**Pull request by ntijoh-daniel-berg**
As discussed in #1205 

Lots of changes since all config.json files were updated. Also patched the test system to work with concept exercises.
Some exercises (such as amusement-park) needed manual renaming of files and "requires" (as in `require_relative`) to work with the test system.

### CODE REVIEW

File: exercises/practice/affine-cipher/.meta/config.json
Comment (kotp) [inline exercises/practice/affine-cipher/.meta/config.json:25]: A `JSON` file should end with an EOL as it is a text file.

**Review by kotp — COMMENTED** [review]
The JSON files consistently are missing the last line.  That is to say, there is an EOL (end of line) character 0x0a at the last line, followed by additional data (text) at what should be the last line.

File: exercises/practice/affine-cipher/.meta/config.json
Comment (ntijoh-daniel-berg) [reply exercises/practice/affine-cipher/.meta/config.json]: Aha, I did not know that.  Fixed.
`configlet lint` now passes locally (but with warnings I think are outside the scope of this fix ).

**Comment by kotp** [comment]
Trailing white space is definitely within the scope of this issue, otherwise it only creates more noise later.

The `has_pass?` is not idiomatic in Ruby, the `has_` being redundant in the `?` part of the name.  Personally, I would exempt the naming via the following comment (see my answer [here](https://stackoverflow.com/a/26350879):

```ruby
def has_pass? # rubocop:disable Naming/PredicateName
```

In this way, it signals to the student that the name is not idiomatic.  But also alerts the reader of that code (from the maintainers standpoint) that we might want to attend to the exercise that is causing this to happen.

**Review by kotp — APPROVED** [review]

**Comment by kotp** [comment]
Very nice.   Thank you @ntijoh-daniel-berg !

**Comment by ntijoh-daniel-berg** [comment]
Fixed rubocop errors.
I agree w/ the rubocop disable directive.

> Trailing white space is definitely within the scope of this issue, otherwise it only creates more noise later.

Yes, I agree, the linter warnings I was referencing was things like `The following concepts exist in the `prerequisites` array of a Practice Exercise in `... , but do not exist in the top-level `concepts` array`

**Comment by kotp** [comment]
Can you create an issue for those, as reported?  That will put them on the radar, at least.

**Comment by ntijoh-daniel-berg** [comment]
Done.

**Comment by iHiD** [comment]
@ntijoh-daniel-berg Thanks for this.

Note: The changes from `attendee.rb` to `amusement_park_improvements.rb` will result in people getting duplicate files when working in the editor. I'll revert that in a separate PR.

@kotp Please can you tag me in any issues that change things like the `config.json` files, so I can analyze the consequences before we merge. Thanks 🙂

**Comment by kotp** [comment]
> tag me in any issues that change things like the `config.json` files…

Ooh… yeah!

**Comment by kotp** [comment]
@ihid That would be a lovely thing to have happen automatically, I think, if possible.  Any changes to a certain list of files automatically alerting specific users or group of users.  Might be a good feature request for github.

**Comment by iHiD** [comment]
We could add me as a codeowner of `config.json` files. I'm not sure of the syntax to do that recursively. @ErikSchierboom Do you know?

**Pull request merged**
